OsuPlayer/Views/EditUserView.axaml.cs

```diff
@@ -216,14 +186,49 @@ private async void SaveChanges_OnClick(object? sender, RoutedEventArgs e)
         {
             ViewModel.NewPassword = string.Empty;
             ViewModel.Password = string.Empty;
-            
+
             ProfileManager.User = ViewModel.CurrentUser;
-            
+
             if (changedProfilePicture)
                 await UpdateProfilePicture();
         }
     }
 
+    /// <summary>
+    /// This handles the <see cref="UserResponse"/> got from editing the user.
+    /// </summary>
+    /// <param name="response">the response to handle</param>
+    /// <returns>true if the response is a successful one, false otherwise</returns>
+    private async Task<bool> HandleResponse(UserResponse response)
+    {
+        switch (response)
+        {
+            case UserResponse.UserEdited:
+                await MessageBox.ShowDialogAsync(_mainWindow, "Profile edited successfully!");
+                return true;
+            case UserResponse.UserEditedAndPasswordChanged:
+                await MessageBox.ShowDialogAsync(_mainWindow,
+                    "Profile edited successfully and password changed.");
+                return true;
+            case UserResponse.UserEditedAndPasswordNotChanged:
+                await MessageBox.ShowDialogAsync(_mainWindow,
+                    "Profile edited successfully. However your password couldn't be changed!");
+                return true;
+            case UserResponse.UserAlreadyExists:
+                await MessageBox.ShowDialogAsync(_mainWindow,
+                    "Profile not updated because the user already exists. Did you try changing your name?");
+                return false;
+            case UserResponse.PasswordIncorrect:
+                await MessageBox.ShowDialogAsync(_mainWindow,
+                    "The password you entered is not correct, therefore your profile was not updated!");
+                return false;
+            default:
+                await MessageBox.ShowDialogAsync(_mainWindow,
+                    string.Format($"We had trouble updating your profile error message:\n\n{response:G}"));
+                return false;
+        }
+    }
+
```

Comment:
Stuff like this next time in it's own issue, PR or branch. Because this has nothing todo with playlists.

Anyhow. I like the Idea, makes the codebase cleaner.

Reply:
:)